agent/xds/listeners.go

```diff
@@ -227,30 +228,50 @@ func (s *ResourceGenerator) listenersFromSnapshotConnectProxy(cfgSnap *proxycfg.
 		}
 	}
 	requiresTLSInspector := false
+	requiresHTTPInspector := false
 
 	err = cfgSnap.ConnectProxy.DestinationsUpstream.ForEachKeyE(func(uid proxycfg.UpstreamID) error {
 		svcConfig, ok := cfgSnap.ConnectProxy.DestinationsUpstream.Get(uid)
 		if !ok || svcConfig == nil {
 			return nil
 		}
-
-		for _, address := range svcConfig.Destination.Addresses {
-			clusterName := clusterNameForDestination(cfgSnap, uid.Name, address, uid.NamespaceOrDefault(), uid.PartitionOrDefault())
+		configuredPorts := make(map[int]interface{})
```

Comment:
This is created on each key, so I think it's always going to be empty.

Also, do we ever write the ports we have "seen"?

Reply:
yeah I missed some of the logic

Reply:
I changed the logic to fix that, please take another look